Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -45,100 +46,83 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
 
-                    string edOrgTypes = string.Join(
-                        "', '",
-                        authorizationSegment.ClaimsEndpoints
-                            .Select(s => s.Name.TrimSuffix("Id"))
-                            .Distinct()
-                            .OrderBy(x => x));
+                // This should never happen
+                if (subjectEndpointWithValue == null)
+                {
+                    throw new Exception(
+                        "The claims-based authorization segment subject endpoint for a single-item authorization was not defined with a value.");
+                }
 
+                if (subjectEndpointWithValue.Value == null)
+                {
                     throw new EdFiSecurityException(
-                        $"Unable to authorize the request because there is no authorization support for associating the "
-                        + $"API client's associated education organization types ('{edOrgTypes}') with the resource.");
+                        $"Access to the resource item could not be authorized because the '{subjectEndpointWithValue.Name}' of the resource is empty.");
                 }
 
-                // Combine multiple statements resulting from multiple EdOrg types in the API client's claims using "OR"
-                // (This forces at least 1 of the relationships with the EdOrg types to exist)
-                segmentStatements.Add(string.Join($"{Environment.NewLine}OR ", segmentExpressions));
+                // Perform defensive checks against the remote possibility of SQL injection attack
+                ValidateTableNameParts(claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
+
+                string derivedAuthorizationViewName = ViewNameHelper.GetFullyQualifiedAuthorizationViewName(
+                    subjectEndpointName,
+                    claimEndpointName,
+                    authorizationSegment.AuthorizationPathModifier);
+
+                if (!IsAuthorizationViewSupported(derivedAuthorizationViewName))
+                {
+                    unsupportedAuthorizationViews.Add(derivedAuthorizationViewName);
+
+                    edOrgTypes.AddRange(authorizationSegment.ClaimsEndpoints
+                       .Select(s => s.Name.TrimSuffix("Id")));                    
+
+                    continue;
+                }
+
+                segmentStatements.Add(CreateSegmentExpression(ref parameterIndex));
+
+                string CreateSegmentExpression(ref int index)
+                {
+                    return string.Format(
+                        StatementTemplate,
+                        GetMultiValueCriteriaExpression(authorizationSegment.ClaimsEndpoints.ToList(), parameters, ref index),
+                        subjectEndpointName.ContainsIgnoreCase("EducationOrganizationId")
+                        ? "TargetEducationOrganizationId" : subjectEndpointName,
```

Comment:
If this line is to be broken like this, `?` needs to be indented and `:` needs to be on the next line, aligned with the `?` please.